### PR TITLE
Centralize and type no_default

### PR DESCRIPTION
**Pull request by crusaderky**
- Follow-up https://github.com/dask/dask/pull/10495
- Blocked by #8192 
- Blocked by #8193 
- Remove some very old deprecations

### CODE REVIEW

File: distributed/client.py [hunk 1]
```diff
@@ -1451,13 +1447,7 @@ def wait_for_workers(
             Time in seconds after which to raise a
             ``dask.distributed.TimeoutError``
         """
-        if n_workers is no_default:
-            warnings.warn(
-                "Please specify the `n_workers` argument when using `Client.wait_for_workers`. Not specifying `n_workers` will no longer be supported in future versions.",
-                FutureWarning,
-            )
-            n_workers = 0
-        elif not isinstance(n_workers, int) or n_workers < 1:
```
Comment (crusaderky) [inline distributed/client.py:1460]: deprecated since Aug 2022

File: distributed/deploy/cluster.py [hunk 2]
```diff
@@ -634,13 +629,7 @@ def wait_for_workers(
             Time in seconds after which to raise a
             ``dask.distributed.TimeoutError``
         """
-        if n_workers is no_default:
-            warnings.warn(
-                "Please specify the `n_workers` argument when using `Client.wait_for_workers`. Not specifying `n_workers` will no longer be supported in future versions.",
-                FutureWarning,
-            )
-            n_workers = 0
-        elif not isinstance(n_workers, int) or n_workers < 1:
```
Comment (crusaderky) [inline distributed/deploy/cluster.py:643]: Deprecated since Aug 2022

File: distributed/tests/test_utils.py [hunk 3]
```diff
@@ -700,58 +700,10 @@ async def set_var(v: str) -> None:
     await asyncio.gather(set_var("foo"), set_var("bar"))
 
 
-def test_serialize_for_cli_deprecated():
```
Comment (crusaderky) [inline distributed/tests/test_utils.py:703]: All these were deprecated since 2021

File: distributed/utils.py [hunk 4]
```diff
@@ -1605,15 +1603,7 @@ def clean_dashboard_address(addrs: AnyType, default_listen_ip: str = "") -> list
...
 _deprecations = {
-    "deserialize_for_cli": "dask.config.deserialize",
-    "serialize_for_cli": "dask.config.serialize",
-    "format_bytes": "dask.utils.format_bytes",
-    "format_time": "dask.utils.format_time",
-    "funcname": "dask.utils.funcname",
-    "parse_bytes": "dask.utils.parse_bytes",
-    "parse_timedelta": "dask.utils.parse_timedelta",
-    "typename": "dask.utils.typename",
-    "tmpfile": "dask.utils.tmpfile",
```
Comment (crusaderky) [inline distributed/utils.py:1616]: All these were deprecated since 2021

Comment (fjetter) [reply distributed/utils.py:1616]: In the interest of a meaningful changelog, I'm inclined to split off these changes. The title of this PR will end up in the changelog and nothing in there says anything about removing deprecated functionality or configuration

File: distributed/client.py (hunk 1 above)
Comment (fjetter) [reply distributed/client.py:1461]: same here. These should be separate changes

**Review by fjetter — COMMENTED** [review]
I don't mind the changes themselves but I would like us to adopt a better habit of communicating these changes properly. The mechanism right now is to put PR titles into the changelog.

**Comment by crusaderky** [comment]
@fjetter opened #8192 and #8193

**Review by phofl — APPROVED** [review]
good to merge

**Pull request merged**